spec/datadog/core/buffer/shared_examples.rb

```diff
@@ -0,0 +1,297 @@
+# typed: false
+require 'spec_helper'
+
+require 'benchmark'
+require 'concurrent'
+
+RSpec.shared_examples 'thread-safe buffer' do
+  subject(:buffer) { described_class.new(max_size) }
```

Comment:
The contents of this file were moved over from `spec/ddtrace/buffer_spec.rb` but not removed from there. Can you doublecheck if this is intended?

Reply:
Yes, this is intentional. Both examples in `Core` and `ddtrace` need this example group. For lack of inheriting test components from `Core`, I think this should be acceptable to duplicate for now. Something we can always optimize if we have time to tune up the test suite later.

Reply:
Seems reasonable. I'd only suggest calling it out with a comment on both files, so that we're aware of this duplication and the reasoning behind it.